packages/api/src/v3/resourceStates.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Json } from "../types";
+
+export interface CloudResource extends Json {
+  type?: string;
+  resourceId?: string; //resourceId
+  resourceName?: string;
+  endpoint?: string;
+  skuName?: string;
+  secretFields?: string[];
+}
+
+export interface AppResource extends Json {
+  appId: string;
+}
+
+export interface ResourceStates {
+  solution: Json;
+  [key: string]: CloudResource | CloudResource[];
+}
+
+////////////Azure Solution/////////////////////
+export interface AzureResource extends CloudResource {
+  resourceGroupName?: string;
```

Comment:
Is it possible that we reduce optional fields as much as possible?

Reply:
Discussed with Chaoyi, actually, for Azure resourceId already encode the resource group and subscription information, so is deleted